### ISSUE TITLE
Add support for operation headers

### ISSUE THREAD

**Issue by bjwschaap** [biemond/biemond-wildfly#113]
The HTTP management API has so-called `operation-headers`, which allow some resources to be added without requiring the (full) server to reload. Please see:
https://docs.jboss.org/author/display/WFLY9/Management+API+reference#ManagementAPIreference-OperationHeaders
I think we should offer support for these headers, so that some resources can be added more efficiently because then they won't require the server to reload.
Looking at the code I think this feature could actually be implemented easily and quickly..

**Comment by jairojunior** [comment]
That's very interesting. It opens a door for a lot of possibities.

Management API docs used to be really shallow when I started writing this module. Always had to write something in CLI and convert to HTTP API syntax, even had to investigate Management Console interactions with HTTP API to better understand this...

Do you think we should have a custom type that supports plain JSON requests? So people could tests commands using HTTP API before using it in the module.

**Comment by bjwschaap** [comment]
I'm not sure. I'm leaning on two different thoughts regarding this:
- it would be nice to have this as a 'last resort' type to get things done with the module even when there's no specific type for it (this is what I actually use `wildfly::util::resource` for right now)
- on the other hand this would be the `exec` type of Wildfly. And I think using `exec` is bad. Well, at least 99% of the time... An exec can always be replaced with a custom type.

What do you think?

**Comment by jairojunior** [comment]
It's nice when you automate a task and are able to easily transport it to Puppet, but as you suggested, we should stick the simple for now.

Let's just introduce a new parameter for operation headers in our custom types.

:+1:

**Comment by jairojunior** [comment]
Operation headers are now supported with the new CLI Parser implemented in v1.0.0 branch and it should work for `wildfly_cli`. Just need to introduce this parameter for `wildfly_resource` and `wildfly_deployment`.

Not sure whether should use CLI syntax:

```puppet

wildfly_resource { '/resource=name':
  operation_headers => '{header1=value;header2=value}'
}
``` 

Or a typed alternative which I would be able to validate headers name:

```puppet

wildfly_resource { '/resource=name':
  operation_headers => {'rollback-on-runtime-failure' => false}
}
```

**Comment by jairojunior** [comment]
Done for `wildfly_resource` and `wildfly_deployment` and its wrappers. Tested for `wildfly_cli` using `{header1=value;header2=value}` syntax.
